### DISCUSSION TITLE
CMAKE_INSTALL_RPATH

### DISCUSSION THREAD

**Discussion by magnusviri** [pkgxdev/pkgx#488] [Help]
The [homebrew formula for abseil](https://github.com/Homebrew/homebrew-core/blob/c18a35db1a25be4c080c81cde6cdf6617392da1f/Formula/abseil.rb) has a flag for rpath.

```
cmake -S . -B build \
      -DCMAKE_INSTALL_RPATH={{rpath}} \
```

I don't know make (and cmake) that well, so I [looked it up](https://duerrenberger.dev/blog/2021/08/04/understanding-rpath-with-cmake/). This looks like it affects relocatability. So I thought I'd ask what I should do here.

**Comment by jhheider** [comment]
Hopefully, nothing. `fix-machos.rb` and `fix-elf.ts` try to handle `rpath`s correctly for their respective architecture types.

**Comment by magnusviri** [comment]
So, I shouldn't include it? I just noticed my [pull request](https://github.com/teaxyz/pantry/pull/1005) has it set to `-DCMAKE_INSTALL_RPATH={{ prefix }}/bla`. Oops. I better fix that.

**Reply by jhheider** [reply]
Yes, I don't believe it should be necessary.

**Reply by magnusviri** [reply]
Thanks!

**Comment by mxcl** [comment]
Yeah we do relocatability fixes after the builds. This *won’t* fix hardcoded paths in the end-binary. For that we have to do patches. See eg. openssl and git.

But yeah you'll want to set the `-DCMAKE_INSTALL_RPATH` to install during builds.
